### PR TITLE
About page

### PR DESCRIPTION
**Pull request by nataliet57**
Created an about page to contain more information about our implementation, solution development, and system architecture 
Screen Shot 2024-03-11 at 10 14 27 PM
image

### CODE REVIEW

**Review by YvoneYang — APPROVED** [review]

File: pages/about.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                    )
+                ),
+                html.Br(),
+                html.Div(
+                    style={'display': 'flex', 'justify-content': 'center', 'align-items': 'center'},
+                    children=html.Div(
+                        style={'background-color': 'white', 'padding': '1rem 2rem', 'border-radius': '10px', 'width': '590px', 'min-height': '150px', 'text-align': 'center', 'margin-right': '20px'},
+                        children=[
+                            html.H2('Linear Programming Problem (IP)', style={'color': '#19264f'}),
+                            html.P('By integrating an original Integer programming model, we can mathematically identify the coverage of our charging stations. We utilized this model to define a neighborhood around each potential charging station location, considering areas within a certain distance along the shortest path as covered. Our objective was to minimize costs while achieving a predefined coverage level.', style={'color': '#4C4C4C'})
```
Comment (thomaspaenns) [inline pages/about.py:57]:
```suggestion
                            html.P('By integrating an original Integer Programming Model, we determine the optimal locations of charging stations. The objective of the model is to maximize the coverage of segments (defined as fraction of demand met in that segment) while adhering to a budget constraint', style={'color': '#4C4C4C'})
```
Accuracy

File: pages/about.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                        ]
+                    )
+                ),
+                html.Br(),
+                html.Div(
+                    style={'display': 'flex', 'justify-content': 'center', 'align-items': 'center'},
+                    children=html.Div(
+                        style={'background-color': 'white', 'padding': '1rem 2rem', 'border-radius': '10px', 'width': '590px', 'min-height': '150px', 'text-align': 'center', 'margin-right': '20px'},
+                        children=[
+                            html.H2('Data Collection', style={'color': '#19264f'}),
```
Comment (thomaspaenns) [inline pages/about.py:67]:
```suggestion
                            html.H2('Data Problem', style={'color': '#19264f'}),
```
Consistency

File: pages/about.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                            html.P('Our data collection process involved gathering traffic volume data from Ontario\'s Ministry of Transportation, analyzing charging station utilization across the province, and referencing successful charging network projects in the United States, such as the West Coast Pacific Highway initiative. By synthesizing insights from these diverse sources, we gained valuable information to inform our strategic planning efforts for optimizing EV charging infrastructure in Ontario.', style={'color': '#4C4C4C'})
```
Comment (thomaspaenns) [inline pages/about.py:68]:
```suggestion
                            html.P('Our team leveraged public traffic volume and road data from Ontario\'s Ministry of Transportation, along with publicly available US government data on existing charging infrastructure in North America. All data sources were validated by the Ontario Ministry of Transportation. By synthesizing insights from these diverse sources, we were able to define the parameters of our Model and our Simulation.', style={'color': '#4C4C4C'})
```

File: pages/about.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                        ]
+                    )
+                ),
+                html.Br(),
+                html.H3('Event Simulation', id='simulation-flow', style={'font-weight': 'bold'}),
+                html.Div('The simulation primarily tracks the series of events that will occur throughout the process of charging a vehicle. Critical parameters such as driving speed, balking probability due to long queues, and '
```
Comment (thomaspaenns) [inline pages/about.py:74]:
```suggestion
                html.Div('The flow chart below, illustrates the process of any one given vehicle in the simulation. While the simulation is being run, this process occurs thousands of times, across the road network. '
```
Accuracy

File: pages/about.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                         'duration of charging stall service time were assumed based on historical data. Future steps involve large-scale SimPy execution in Python'),
```
Comment (thomaspaenns) [inline pages/about.py:75]:
```suggestion

```
Already said this info

File: pages/about.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    children=html.Div(
+                        style={'background-color': 'white', 'padding': '1rem 2rem', 'border-radius': '10px', 'width': '590px', 'min-height': '150px', 'text-align': 'center', 'margin-right': '20px'},
+                        children=[
+                            html.H2('Data Collection', style={'color': '#19264f'}),
+                            html.P('Our data collection process involved gathering traffic volume data from Ontario\'s Ministry of Transportation, analyzing charging station utilization across the province, and referencing successful charging network projects in the United States, such as the West Coast Pacific Highway initiative. By synthesizing insights from these diverse sources, we gained valuable information to inform our strategic planning efforts for optimizing EV charging infrastructure in Ontario.', style={'color': '#4C4C4C'})
+                        ]
+                    )
+                ),
+                html.Br(),
+                html.H3('Event Simulation', id='simulation-flow', style={'font-weight': 'bold'}),
```
Comment (thomaspaenns) [inline pages/about.py:73]:
```suggestion
                html.H3('Simulation Flow Chart', id='simulation-flow', style={'font-weight': 'bold'}),
```
Consistency

File: pages/about.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                html.Div(
+                    style={'display': 'flex', 'justify-content': 'center', 'align-items': 'center'},
+                    children=html.Img(
+                        src='assets/simulation-flow_chart.png',
+                        style={'max-height': '250px', 'align-items': 'center'}
+                    )
+                ),
+                html.Br(),
+                html.H3('System Flow Diagram', id='system-flow', style={'font-weight': 'bold'}),
+                html.Div('The following diagram illustrates the relationship between the tools, and the FAQ section below could address some of the questions you might have. For each model, certain features will be necessary. For all features, if future projections cannot be obtained then they will have to be forecasted.'),
```
Comment (thomaspaenns) [inline pages/about.py:86]:
```suggestion
                html.Div('The following diagram illustrates the relationship between the components of our tool, and how you as a user might iteratively interact with them. The FAQ section below could address some of the questions you might have. '),
```
Accuracy

File: pages/about.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                html.Div(
+                    style={'display': 'flex', 'justify-content': 'center', 'align-items': 'center'},
+                    children=html.Img(src='assets/final_system.png', style={'align-items': 'center'}),
+                ),
+                html.Br(),
+                html.Br(),
+                html.H3('Frequently Asked Questions', id='faq', style={'font-weight': 'bold'}),
+                html.Div([
+                    html.H5('1. What is Integer Programming (IP)?'),
```
Comment (thomaspaenns) [inline pages/about.py:95]:
```suggestion
                    html.H5('1. What is an Integer Programming Model?'),
```
Clarity

File: pages/about.py [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+                    html.P('IP is a mathematical optimization technique where decision variables are restricted to be integers. Link to our IP problem can be found here:'),
```
Comment (thomaspaenns) [inline pages/about.py:96]:
```suggestion
                    html.P('An integer programming model allows us to define locations of stations as integer decision variables, and define constraints and parameters such that a solver can then find the optimal solution to the defined problem'),
```
Accuracy and Clarity

File: pages/about.py [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+                    html.Div(
+                        [
+                            html.A("Link to our IP problem", href="https://uofwaterloo.sharepoint.com/:b:/s/tm-arts-23-24mgtecapstone-Team10/EWrFsD7LvMlDmAouvAOTAO0BRn96RlvSks6X56_Q0UrURQ?e=5U9hnl")
+                        ]),
+                    html.Br(),
+
+                    html.H5('2. What is M/M/C model in queuing theory?'),
```
Comment (thomaspaenns) [inline pages/about.py:103]:
```suggestion

```
This is defined above

File: pages/about.py [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+                    html.P('An M/M/c model represents a queueing system that has exponential arrivals, c servers with exponential service time'),
```
Comment (thomaspaenns) [inline pages/about.py:104]:
```suggestion

```
This is also defined above

File: pages/about.py [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+                    html.H5('3. What do the charging ports mean?'),
```
Comment (thomaspaenns) [inline pages/about.py:106]:
```suggestion
                    html.H5('3. What do the numbers of charging ports represent?'),
```

File: pages/about.py [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+                    html.P('The 2, 4, 8 charging ports in the simulation refers to the number of physical connectors or outlets available for vehicles to plug into and charge simultaneously at a charging station.'),
```
Comment (thomaspaenns) [inline pages/about.py:107]:
```suggestion
                    html.P('The program allows both the user and/or the optimization model to select either 2, 4, or 8 charging ports for any given charging station. This refers to the number of physical connectors or outlets available for vehicles to plug into and charge simultaneously at a charging station.'),
```
Accuracy and Clarity

File: pages/about.py [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+                    html.H5('4. Is Voltpath suitable for large-scale deployment planning?'),
+                    html.P('Yes, Voltpath is designed to handle large-scale deployment planning for EV charging stations. Its optimization and simulation capabilities make it suitable for strategic planning at the regional or city level.'),
```
Comment (thomaspaenns) [inline pages/about.py:110]:
```suggestion
                    html.P('Yes, Voltpath is designed to handle large-scale deployment planning for EV charging stations. Its optimization and simulation capabilities make it suitable for strategic planning at the regional or provincial level.'),
```
Accuracy

File: pages/about.py [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+                html.Ul([
+                    html.Li(html.A('Purpose of Voltpath', href='#purpose')),
+                    html.Li(html.A('Solution Development', href='#solution')),
+                    html.Li(html.A('System Flow Diagram', href='#system-flow')),
+                    html.Li(html.A('Simulation Flow', href='#simulation-flow')),
+                    html.Li(html.A('FAQ', href='#faq')),
+                ]),
+                html.Br(),
+            ]),
+            width=3
```
Comment (thomaspaenns) [inline pages/about.py:22]: Can you make this column slightly less wide?

Comment (nataliet57) [reply pages/about.py:22]: Set it to 2!

**Review by thomaspaenns — APPROVED** [review]
LGTM!

**Pull request merged**